src/storage/ducklake_metadata_manager.cpp

```diff
@@ -1503,21 +1515,66 @@ void DuckLakeMetadataManager::WriteNewSchemas(DuckLakeSnapshot commit_snapshot,
 	}
 }
 
+string GetExpressionType(ParsedExpression &expression) {
+	switch (expression.type) {
+	case ExpressionType::OPERATOR_CAST: {
+		auto &cast_expression = expression.Cast<CastExpression>();
+		if (cast_expression.child->type == ExpressionType::VALUE_CONSTANT) {
+			return "literal";
+		}
+		return "expression";
+	}
+	case ExpressionType::FUNCTION:
+	case ExpressionType::COLUMN_REF:
+		return "expression";
+	case ExpressionType::VALUE_CONSTANT:
```

Comment:
Can this be `default: return "expression"`? This list seems unnecessarily restrictive - e.g. currently we can't have a default `CASE` expression or comparison in the root